appengine/urlfetch/main.py

```diff
@@ -0,0 +1,95 @@
+# Copyright 2016 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Sample application that demonstrates different ways of fetching
+URLS on App Engine
+"""
+
+import urllib
+import urllib2
+import webapp2
+
+from google.appengine.api import urlfetch
+
+
+def handle_error(e):
+    pass
+
+
+class UrlLibFetchHandler(webapp2.RequestHandler):
+    """ Demonstrates an HTTP query using urllib2
+    """
+
+    def get(self):
+        url = "http://www.google.com/"
+        try:
+            result = urllib2.urlopen(url)
+            self.response.write(result.read())
+        except urllib2.URLError, e:
+            handle_error(e)
+
+
+class UrlFetchHandler(webapp2.RequestHandler):
+    """ Demonstrates an HTTP query using urlfetch
+    """
+
+    def get(self):
+        url = "http://www.google.com/"
+        try:
+            result = urlfetch.fetch(url)
+            if result.status_code == 200:
+                self.response.write(result.content)
+            else:
+                self.response.status_code = result.status_code
+        except urllib2.URLError, e:
+            raise e
```

Comment:
might as well log the error if you're going to catch it and not do anything with it.